test/problems/test_linear_programs.jl

```diff
@@ -22,31 +40,26 @@
     #   X'(c - A'y) = 0     [complementarity slackness]
     #
 
-    function assert_lp_solution(c, A, b, x, y, tol)
-        # Check and print solution quality measures (for some reason the
-        # returned dual iterate is the negative of the dual LP variable y)
-
-        nonneg = -minimum(min.(0.0, x))
-        @test nonneg <= tol
-
-        primal_feasibility = norm(A * x - b)
-        @test primal_feasibility <= tol
-
-        dual_feasibility = maximum(max.(0.0, -A' * y - c))
-        @test dual_feasibility <= tol
-
-        complementarity = abs(dot(c + A' * y, x))
-        @test complementarity <= tol
-    end
-
-    n = 100 # primal dimension
-    m = 80 # dual dimension (i.e. number of linear equalities)
-    k = 50 # number of active dual constraints (must be 0 <= k <= n)
-    x_star = vcat(rand(T, k), zeros(T, n - k)) # primal optimal point
-    s_star = vcat(zeros(T, k), rand(T, n - k)) # dual optimal slack variable
-    y_star = randn(T, m) # dual optimal point
-
-    A = randn(T, m, n)
+    n = 10 # primal dimension
+    m = 8 # dual dimension (i.e. number of linear equalities)
+    k = 5 # number of active dual constraints (must be 0 <= k <= n)
+    # primal optimal point
+    x_star = T[0.03606099647643202, 0.6641306619990367, 0.14689326835593086, 0.8616058527226432, 0.6518888386753204, 0.0, 0.0, 0.0, 0.0, 0.0]
```

Comment:
I wonder if it makes sense to use something like `randn(seed, ...)` instead of copy pasting random numbers?

Reply:
The problem is that the random number generator can change across Julia versions, so the seed is not enough. One could use some stable algorithm, decoupled from Julia Base, but I think in the end having a fixed problem like this is not too bad…

Reply:
true, but I wonder if the base library Random has now reached enough stability